src/extension.ts

```diff
@@ -1,106 +1,61 @@
-
-						if (showProcessFullVersion)
-							title += ` ${r.processVersion}`;
-						else {
-							const verisonNumbers = r.processVersion.split('.');
-							title += ` ${verisonNumbers[0]}.${verisonNumbers[1]}`;
-						}
-
-						if (showProcessId)
-							title += ` <${r.processId}>`;
-
-						if (showActiveDocument) {
-							if (r.activeDoc.title == undefined) {
-								title += ` - "Untitled"`;
-							}
-							else {
-								title += ` - "${r.activeDoc.title}"`;
-							}
-						}
-
-						if (showActiveDocumentPath) {
-							if (r.activeDoc.title == undefined) {
-								title += ` - Not Saved`;
-							}
-							else {
-								title += ` - @ ${r.activeDoc.location}`;
-							}
-						}
-
-						if (showActiveViewport)
-							title += ` [${r.activeViewport}]`;
-
-						if (showProcessAge) {
-							let age = timespan.parse(`${r.processAge}m`);
-							title += ` (${timespan.getString(age, { abbv: false })})`;
-						}
-
-						return [title, r]
-					})
-				);
-
-				vscode.window.showQuickPick([...names.keys()], { canPickMany: false })
-					.then((v) => {
-						if (v != undefined) {
-							const rhino = names.get(v);
-							const scriptCmd = `"${getRhinoCode()}" --rhino ${rhino?.pipeId} script \"${activeFile}\"`;
-							console.log(scriptCmd)
-							cp.execSync(scriptCmd);
-						}
-					});
-			}
-		}
-		else {
-			vscode.window.showInformationMessage(
-				"There are no active scripts open"
-			);
-		}
-	});
+import * as Vsc  from 'vscode';
+
+import * as Cp   from 'child_process';
+import * as Os   from 'os';
+import * as Path from 'path';
+import * as Fs   from 'fs'
+import * as Ver  from 'compare-versions';
+
+const Timespan = require("timespan-parser");
+
+
+// Public
+
+/** this method is called when extension is activated */
+export function activate (context: Vsc.ExtensionContext) {
+	let disposable = Vsc.commands.registerCommand ('rhinocode.runInRhino', runInRhinoCommand);
 
-	context.subscriptions.push(disposable);
+	context.subscriptions.push (disposable);
 }
 
-// this method is called when extension is deactivated
-export function deactivate() { }
+/** this method is called when extension is deactivated */
+export function deactivate () { }
+
+
+// Private
+
+const RHC_MIN_VER = "0.2.0";
+const {
+	defaultFolder  : RHC_DEFAULT_FOLDER,
+	relativeBinPath: RHC_RELATIVE_BIN_PATH
+} = (function () {
+	switch (Os.platform ()) {
+		case 'darwin':
+			return {
+				defaultFolder: "/Applications/RhinoWIP.app",
+				relativeBinPath: "Contents/Resources/bin/rhinocode"
+			}
+		case 'win32':
+			return {
+				defaultFolder: "C:\\Program Files\\Rhino 8 WIP\\System",
+				relativeBinPath: "RhinoCode.exe"
+			}
+		default:
+			console.error ("[RhinoCode] Invalid platform:", Os.platform ())
+			return {
+				defaultFolder: "",
+				relativeBinPath: ""
+			}
+	}
+})()
 
 interface RhinoInstance {
-	readonly pipeId: string;
-	readonly processId: number;
-	readonly processName: string;
+	readonly pipeId        : string;
```

Comment:
My vscode typescript formatter removes these spaces. Are you using a different formatter?

Reply:
Ah, yes, I haven't changed that.
It is not better, it is not less good. it is quite possible to put the names back as before.

I write the modules with a capital letter because of `path` (essentially).
`path` is very often a variable `path = "dir/file.txt" `, but it is also very often the module `const path = require ('path') `
I have had bugs several times because of this.

Between `cp` and `Cp` we can identify the variable `cp` of the module `Cp`.